### PR TITLE
Add GEP describing goals for service mesh in Gateway API

### LINKED ISSUE #1324: GEP: Service Mesh in Gateway API
**What would you like to be added**:
Due to the overlap in surface area between the Gateway API's routing primitives and many service mesh implementations, it would be generally beneficial to the Kubernetes community for the Gateway API to provide/support patterns and/or new resources for service mesh functionality. 

**Why this is needed**:
While attempting to scope out #1291 and #1294, it became apparent that more higher level context was needed to first answer the "what" question. Both of those issues are intrinsically tied to implementation details, so it is useful to first have a GEP that describes what service mesh interoperability in Gateway API is trying to accomplish.

### PR DESCRIPTION
**Pull request by keithmattix**
Signed-off-by: Keith Mattix II <keithmattix2@gmail.com>

**What type of PR is this?**

/kind gep

**What this PR does / why we need it**:
This PR adds a GEP describing the high level goals for adding service mesh functionality in Gateway API as well as a glossary of terms to aid in discussion

**Which issue(s) this PR fixes**:

Fixes #1324

**Does this PR introduce a user-facing change?**:

```release-note
Add GEP describing goals for service mesh in Gateway API
```

### CODE REVIEW

**Review by howardjohn — COMMENTED** [review]
/lgtm

Thanks @keithmattix !

File: site-src/geps/gep-1324.md
Comment (shaneutt) [inline site-src/geps/gep-1324.md:8]: Good summary :+1:

File: site-src/geps/gep-1324.md
Comment (shaneutt) [inline site-src/geps/gep-1324.md:12]:
```suggestion
* Augment the Gateway API specification with the necessary models, resources, and modifications so that the xRoute primitives are usable by service mesh implementations
```

File: site-src/geps/gep-1324.md
Comment (shaneutt) [inline site-src/geps/gep-1324.md:13]: I think we need to elaborate on this one, can we make this goal higher level, or explain it further so there's little or no ambiguity?

File: site-src/geps/gep-1324.md
Comment (shaneutt) [inline site-src/geps/gep-1324.md:18]: I think all of these goals are solid and read clearly to me :+1:

File: site-src/geps/gep-1324.md
Comment (shaneutt) [inline site-src/geps/gep-1324.md:24]: Good non-goals :+1:

File: site-src/geps/gep-1324.md
Comment (shaneutt) [inline site-src/geps/gep-1324.md:32]: Small suggestion about prominence and language: consider moving this to the bottom of the GEP under "Alternatives Considered" and use language that clearly indicates we mean not to cover them in _this_ GEP, but instead of a future GEP may introduce them.

File: site-src/geps/gep-1324.md
Comment (shaneutt) [inline site-src/geps/gep-1324.md:63]: :+1:

File: site-src/geps/gep-1324.md
Comment (shaneutt) [inline site-src/geps/gep-1324.md:117]: This is a _very_ important distinction which [we've touched on before](https://github.com/kubernetes-sigs/gateway-api/pull/993). I think it's fine to add this to the glossary as it's purely contextual, but as someone who has advocated to make this distinction within Gateway APIs in the past (coincidentally, specifically in order to support service mesh use cases with Gateway API) I can see this part in particular getting tricky. Nothing to do about it right now, but consider that the distinction of frontend vs backend with a Kubernetes `Service` may very well be something that could use its own GEP entirely.

**Review by shaneutt — APPROVED** [review]
I think this is a _great_ starting point and it did a really great job of presenting the "what" and the "why" to me without making me worry yet about the how. I also particularly liked the glossary. As such, and because this is `provisional` and I feel we should aim to merge fast during the provisional stage I'm approving this as I do not feel there's anything worth blocking on.

I do have a few comments on the PR: I would ask that for the ones that specifically request action they be resolved before we merge, and for all other comments please resolve them at your discretion.

I'll leave the LGTM for another maintainer so we have one more pair of :eyes: on it.

Great job @keithmattix I'm excited for us to continue down this path :vulcan_salute:

File: site-src/geps/gep-1324.md
Comment (keithmattix) [reply site-src/geps/gep-1324.md:32]: Are these really "alternatives" though? It feels more like scope we're shedding or deferring until later, hence the term "deferred goals". I do agree with moving it to the bottom though

File: site-src/geps/gep-1324.md
Comment (keithmattix) [reply site-src/geps/gep-1324.md:13]: This goal originated from this Google Docs [comment](https://docs.google.com/document/d/1IzItdX22fHfXCP6Zch6pC8hEImMNJ6MCPqGNW1aaDr8/edit?disco=AAAAeBMkHBw), but reading back through it, I think it's covered by the non-goal around duplicating concerns. So, I feel like this could probably be removed?

File: site-src/geps/gep-1324.md
Comment (keithmattix) [reply site-src/geps/gep-1324.md:117]: 100% agree; the decoupling of the `Service` resource in the general Gateway API conceptual model probably makes a lot of sense. I don't think I could get behind moving away from the service resource, but clarifying the different functions of the resource is absolutely a worthwhile exercise

File: site-src/geps/gep-1324.md
Comment (shaneutt) [reply site-src/geps/gep-1324.md]: ok :+1:

File: site-src/geps/gep-1324.md
Comment (shaneutt) [reply site-src/geps/gep-1324.md]: I can compromise on the `alternatives` bit, the bottom seems good.

File: site-src/geps/gep-1324.md
Comment (shaneutt) [reply site-src/geps/gep-1324.md:108]: Well, I can't think of a good action for the moment so I'll consider this one resolved but I definitely think this is going to be a point of great interest soon.

**Review by shaneutt — APPROVED** [review]

**Comment by youngnick** [comment]
I think this is ready for merging as provisional.

/lgtm

File: site-src/geps/gep-1324.md
Comment (mikemorris) [inline site-src/geps/gep-1324.md:136]:
```suggestion
* Egress (e.g. off-mesh destinations including legacy services or third-party APIs)
```
I think this is currently worded a bit confusingly, and assumes an egress gateway architecture versus allowing a sidecar egress pattern - would honestly just shorten this substantially and plan for an entire egress GEP at some future point.

File: site-src/geps/gep-1324.md
Comment (mikemorris) [inline site-src/geps/gep-1324.md:139]: 👏 Was gonna suggest adding these then saw they made it in already!

File: site-src/geps/gep-1324.md
Comment (keithmattix) [reply site-src/geps/gep-1324.md:56]: Yeah, mkdocs (which renders the website) is based on a stricter implementation of the Markdown spec than GitHub. The extra indentation helps it render correctly in HTML

File: site-src/geps/gep-1324.md
Comment (keithmattix) [reply site-src/geps/gep-1324.md:136]: Agreed; making this change in a follow up PR

File: site-src/geps/gep-1324.md
Comment (keithmattix) [reply site-src/geps/gep-1324.md:48]: It probably depends on the org....skipping way ahead to implementation, the base case for this probably starts as a per-service config setting, so I'll likely keep it here for now

**Pull request merged**
